packages/core/src/utils/stripTypename.ts

```diff
@@ -0,0 +1,22 @@
+export const stripTypename = (data: any): any => {
+  if (!data) return data;
+
+  return Object.keys(data).reduce((acc, key: string) => {
+    const value = data[key];
+    if (key === '__typename') {
+      return acc;
+    } else if (Array.isArray(value)) {
+      return {
+        ...acc,
```

Comment:
We don't need to spread acc here to avoid excessive object allocations and can just mutate (as before).

Does it make sense to add `__typename` as `Object.defineProperty({ value, enumerable: false, writable: true, configurable: true })`?